packages/platform/src/services/PlayService.ts

```diff
@@ -202,7 +202,7 @@ interface GetPlayerResultArgs {
 }
 
 export async function getPlayerResult(args: GetPlayerResultArgs, ctx: Context) {
-  const currentGame = await ctx.prisma.game.findUnique({
+  let currentGame = await ctx.prisma.game.findUnique({
```

Comment:
I see your point. However, having an immutable data flow often comes to having a cost of copying. That given, is it worth to copy for every query? 

So, I assume what you suggest would be:
```
  const currentGame = await ctx.prisma.game.findUnique({ ... })
  const modifiedCurrentGame = {
    ...currentGame,
    periods: currentGame.periods.map(...).filter(....),
    activePeriod: {
    ...currentGame.activePeriod,
    segments: currentGame.activePeriod.filter(...)
    }
  }
  // Another copy as we want to update currentGame.periods[currentGame.activePeriodIx].segments ...
```
Did I understand you correctly? Or would you suggest to use immer here as well (which would be than the above)?

If the scope is short and it is clear what the code is doing (in this case only a modified query without further modifications) I am usually fine with modifying the output. Otherwise, I would suggest, that we do it directly at the DB side with _count etc. so it's clear what we get from the database and without mutations.